### PR TITLE
Fix Non-Exclusive Behaviour of `identity_group_policies`

### PR DESCRIPTION
**Pull request by lawliet89**
This PR uses the same utility functions added in https://github.com/terraform-providers/terraform-provider-vault/pull/488 and should not result in any merge conflicts.

### CODE REVIEW

**Comment by tyrannosaurus-becks** [comment]
Hi @lawliet89 ! Weirdly, GH is still showing those utility functions in the diff even though I merged them to master. Would you be willing to pull master and push it up to fix the diff?

**Comment by lawliet89** [comment]
@tyrannosaurus-becks I have done a rebase with master.

File: vault/resource_identity_group.go [hunk 1]
```diff
@@ -249,23 +249,28 @@ func identityGroupIDPath(id string) string {
...
-	if resp, err := readIdentityGroup(client, groupId); err != nil {
-		return nil, fmt.Errorf("error reading IdentityGroup policies %q: %s", groupId, err)
-	} else {
-		presentPolicies = resp.Data["policies"].([]interface{})
+func readIdentityGroupPolicies(client *api.Client, groupID string) ([]interface{}, error) {
+	resp, err := readIdentityGroup(client, groupID)
+	if err != nil {
+		return nil, err
+	}
+	if resp == nil {
```
Comment (tyrannosaurus-becks) [inline vault/resource_identity_group.go]: Nice catch here that the response can be nil.

File: vault/resource_identity_group.go [hunk 2]
```diff
@@ -249,23 +249,28 @@ func identityGroupIDPath(id string) string {
...
+	if v, ok := resp.Data["policies"]; ok && v != nil {
+		return v.([]interface{}), nil
+	}
+	return make([]interface{}, 0), nil
 }
 
-func readIdentityGroup(client *api.Client, groupId string) (*api.Secret, error) {
-	path := identityGroupIDPath(groupId)
-	log.Printf("[DEBUG] Reading IdentityGroup %s from %q", groupId, path)
+func readIdentityGroup(client *api.Client, groupID string) (*api.Secret, error) {
```
Comment (tyrannosaurus-becks) [inline vault/resource_identity_group.go]: Might be nice to add a comment just above this function letting folks know that the resp can be nil.

File: vault/resource_identity_group_policies.go [hunk 3]
```diff
@@ -128,11 +91,7 @@ func identityGroupPoliciesRead(d *schema.ResourceData, meta interface{}) error {
 
 	resp, err := readIdentityGroup(client, id)
 	if err != nil {
-		// We need to check if the secret_id has expired
-		if util.IsExpiredTokenErr(err) {
```
Comment (tyrannosaurus-becks) [inline vault/resource_identity_group_policies.go]: Should we strip this out?

Comment (lawliet89) [reply vault/resource_identity_group_policies.go]: It doesn't really make sense to be here. I think it was just a copy/paste from another resource by the original author. If the user's Vault Token has expired, the whole operation should fail anyway.

File: vault/resource_identity_group.go (hunk 2 above)
Comment (lawliet89) [reply vault/resource_identity_group.go]: Done.

**Pull request merged**